downloader/src/main/kotlin/vcs/GitWorkingTree.kt

```diff
@@ -94,8 +98,18 @@ open class GitWorkingTree(workingDir: File, vcsType: VcsType) : WorkingTree(work
             listSubmodulePaths(it).associateWith { path ->
                 GitWorkingTree(it.workTree.resolve(path), vcsType).getInfo()
             }
+        }.mapValues { it.value.replaceUrlPrefixes() }
+
+    private fun VcsInfo.replaceUrlPrefixes(): VcsInfo {
+        val patchedUrl = repositoryUrlPrefixReplacements.entries.fold(url) { url, (prefix, replacement) ->
+            if (url.startsWith(prefix)) {
+                "$replacement${url.removePrefix(prefix)}"
+            } else url
```

Comment:
Sorry for not noticing earlier, but this could also be

    url.withoutPrefix(prefix)?.let { "$replacement$it" } ?: url

Reply:
This is optional, though, and I'll approve anyway.